a3fe/tests/test_engine_configuration.py

```diff
@@ -0,0 +1,245 @@
+"""Unit and regression tests for the SomdConfig class."""
+
+from tempfile import TemporaryDirectory
+import os
+import logging
+
+from a3fe import SomdConfig
+
+def create_test_input_dir():
+    """Create a temporary directory with a mock somd.cfg file."""
+    temp_dir = TemporaryDirectory()
+    with open(os.path.join(temp_dir.name, "somd.cfg"), "w") as f:
+        f.write("ncycles = 1000\n")
+    return temp_dir
+
+def test_create_config():
+    """Test that the config can be created with different parameters."""
+    with create_test_input_dir() as input_dir:
+        # Test with integer runtime
+        config = SomdConfig(
+            runtime=1,  # Integer runtime
+            input_dir=input_dir
+        )
+        assert isinstance(config, SomdConfig)
+        
+        # Test with float runtime
+        config = SomdConfig(
+            runtime=0.3,  # Float runtime
+            input_dir=input_dir
+        )
+        assert isinstance(config, SomdConfig)
+        
+        # Test with custom stream_log_level
+        config = SomdConfig(
+            runtime=1,
+            input_dir=input_dir,
+            stream_log_level=logging.DEBUG
+        )
+        assert config._stream_log_level == logging.DEBUG
+
+def test_config_yaml_save_and_load():
+    """Test that the config can be saved to and loaded from YAML."""
+    with create_test_input_dir() as input_dir:
+        with TemporaryDirectory() as dirname:
+            config = SomdConfig(
+                runtime=1,  # Integer runtime
+                input_dir=input_dir
+            )
+            yaml_path = os.path.join(dirname, "config.yaml")
+            config.dump(yaml_path)
+            config2 = SomdConfig.load(yaml_path)
+            assert config == config2
+
+def test_get_somd_config():
+    """Test that the SOMD configuration file is generated correctly."""
+    with create_test_input_dir() as input_dir:
+        with TemporaryDirectory() as dirname:
+            config = SomdConfig(
+                integrator="langevinmiddle",
+                nmoves=25000,
+                timestep=4.0,
+                runtime=1,  # Integer runtime
+                input_dir=input_dir,
+                cutoff_type="PME",
+                thermostat=False
+            )
+            config_path = config.get_somd_config(
+                run_dir=dirname,
+                config_name="test"
+            )
+            assert config_path == os.path.join(dirname, "test.cfg")
+
+            expected_config = (
```

Comment:
Might be worth copy an old somd.cfg file and check that we can generate exactly the same thing with the new class.